src/combinatorics/simple_graph/ends/defs.lean

```diff
@@ -0,0 +1,230 @@
+/-
+Copyright (c) 2022 Anand Rao, Rémi Bottinelli. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Anand Rao, Rémi Bottinelli
+-/
+import combinatorics.simple_graph.basic
+import combinatorics.simple_graph.connectivity
+import data.set_like.basic
+import category_theory.category.basic
+import category_theory.filtered
+import topology.category.Top.limits
+
+/-!
+# Ends
+
+This file contains a definition of the ends of a simple graph, as sections of the inverse system
+assigning, to each finite set of vertices, the connected components of its complement.
+-/
+
+universes u
+variables {V : Type u} (G : simple_graph V) (K L L' M : set V)
+
+namespace simple_graph
+
+/-- The components outside a given set of vertices `K` -/
+@[reducible] def comp_out := (G.induce Kᶜ).connected_component
```

Comment:
How about `component_compl` as the name?

Reply:
I'm OK with that. I guess the most meaningful name would be something like `connected_component_compl`, which is a mouthful. Keeping a bit of each of those words is still too long, so `component_compl` makes sense to me. @YaelDillies ?